packages/react-toolbar/Spec.md

```diff
@@ -2,62 +2,149 @@
 
-_Provide some representative example code that uses the proposed API for the component_
+v0 `Toolbar` has support to children API with static components in `Toolbar` as `Toolbar.Button` etc...
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+The only layout variation is size differences, there are 2 sizes `medium` which is the default and `small`.
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+The `Toolbar` will implement a `children` based API and will leverage the use of `context` in the interaction and data flows of child components. To achieve `overflow` items once used inside `CommandBar` the component will leavarage from [priority-overflow](https://github.com/ling1726/priority-overflow)
 
-## Structure
+### Toolbar
 
-- _**Public**_
-- _**Internal**_
-- _**DOM** - how the component will be rendered as HTML elements_
+The root level component serves as a simple container for all possible ToolbarComponents
+
+```typescript
+type ToolbarProps = {
+  /**
+   * Defines toolbar size
+   * @default medium
+   */
+  size?: 'small' | 'medium';
+};
+```
+
+### ToolbarButton
+
+It serves as an override on top of `Button` limiting the possible props only to `size`, `appearance`, `disabled` and `disabledFocusable`
+
+```typescript
+type type ToolbarButtonProps = ComponentProps<ButtonSlots> &
+  Partial<Pick<ButtonProps, 'disabled' | 'disabledFocusable'>> & {
+
+  /**
+   * A button can have its content and borders styled for greater emphasis or to be subtle.
+   * - 'primary': Emphasizes the button as a primary action.
+   * - 'subtle': Minimizes emphasis to blend into the background until hovered or focused.
+   */
+  appearance?: 'primary' | 'subtle';
+;
+```
+
+### ToolbarDivider
+
+It's based on the `Divider` containing few styles overrides
+
+```typescript
+type ToolbarDividerProps = ComponentProps<Partial<DividerSlots>>;
+```
+
+### ToolbarToggleButton
+
+It serves as an override on top of `ToggleButton` limiting the possible props only to `size`, `appearance`, `disabled` and `disabledFocusable`
+
+```typescript
+type ToolbarToggleButtonProps = ComponentProps<Partial<ToggleButtonSlots>>;
+```
 
-## Migration
+### ToolbarRadioGroup
 
-_Describe what will need to be done to upgrade from the existing implementations:_
+It's based on the `RadioGroup` containing few styles overrides
+
+```typescript
+type ToolbarRadioGroupProps = ComponentProps<Partial<RadioGroupSlots>>;
+```
+
+### ToolbarRadio
+
+It's based on the `Radio` containing few styles overrides
+
+```typescript
+type ToolbarRadioProps = ComponentProps<Partial<RadioSlots>>;
+```
+
+### ToolbarCheckbox
+
+It's based on the `Checkbox` containing few styles overrides
+
+```typescript
+type ToolbarCheckboxProps = ComponentProps<Partial<CheckboxSlots>>;
+```
+
+### ToolbarItemGroup
```

Comment:
Could you add more detail for `ToolbarItemGroup` ?

* Is it optional ?
* How does it work with `ToolbarRadioGroup`  ?

Reply:
Sure thing, will do